app/plugin_key_value_store.go

```diff
@@ -47,6 +47,27 @@ func (a *App) SetPluginKeyWithExpiry(pluginId string, key string, value []byte,
 	return nil
 }
 
+func (a *App) CompareAndSetPluginKey(pluginId string, key string, oldValue, newValue []byte) (bool, *model.AppError) {
+	kv := &model.PluginKeyValue{
+		PluginId: pluginId,
+		Key:      key,
+		Value:    newValue,
+	}
+
+	updated, err := a.Srv.Store.Plugin().CompareAndUpdate(kv, oldValue)
+	if err != nil {
+		mlog.Error("Failed to update plugin key value", mlog.String("plugin_id", pluginId), mlog.String("key", key), mlog.Err(err))
+		return updated, err
```

Comment:
If an error occurred, the value should not haven been changed.

```suggestion
		return false, err
```

Reply:
True, `updated` _should_ be `false` in this case. The only reason i'm using `updated` here is then all the control logic lives in one spot which is `PluginStore.CompareAndUpdate(kv *model.PluginKeyValue, oldValue []byte)`, all the clients are just forwarding what's returned. Otherwise each layer has to make a decision on what should return in certain scenarios. Thoughts?

Reply:
Makes scene :+1: